src/app/cli/src/graphql.ml

```diff
@@ -8,6 +8,14 @@ open Auxiliary_database
 
 let result_of_exn f v ~error = try Ok (f v) with _ -> Error error
 
+let result_of_or_error ?error v =
+  Result.map_error v ~f:(fun internal_error ->
+      match error with
```

Comment:
both match cases have `Error.to_string_hum internal_error`, so you could compute that once and use it where needed, instead of duplicating code